web/src/actions/activities.js

```diff
@@ -21,7 +22,20 @@ export const UPDATE_ACTIVITY = 'UPDATE_ACTIVITY';
 const actionWithYears = (type, other) => (dispatch, getState) =>
   dispatch({ type, ...other, years: getState().apd.data.years });
 
-export const addActivity = () => actionWithYears(ADD_ACTIVITY);
+export const addActivity = () => (dispatch, getState) => {
+  const oldActivities = getState().activities.allKeys;
+
+  dispatch(actionWithYears(ADD_ACTIVITY));
+  dispatch(updateBudget());
+
+  const newKey = getState().activities.allKeys.filter(
+    k => !oldActivities.includes(k)
+  )[0];
+  dispatch({
+    type: ADD_ACTIVITY_DIRTY,
+    data: getState().activities.byKey[newKey]
+  });
```

Comment:
Sends a separate action to trigger the dirty reducer when we add an activity, because it wants access to the new activity state - but the new activity state isn't available until after the `ADD_ACTIVITY` action is processed.